plugins/inputs/intel_pmt/intel_pmt.go

```diff
@@ -0,0 +1,400 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux && amd64
+
+package intel_pmt
+
+import (
+	_ "embed"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"html"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"github.com/PaesslerAG/gval"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+var hexToDecRegex = regexp.MustCompile(`0x[0-9a-fA-F]+`)
+
+const (
+	defaultPmtBasePath = "/sys/class/intel_pmt"
+	pluginName         = "intel_pmt"
+)
+
+type pmtFileInfo []fileInfo
+
+type fileInfo struct {
+	path     string
+	numaNode string
+}
+
+type IntelPMT struct {
+	PmtSpec        string          `toml:"spec"`
+	DatatypeFilter []string        `toml:"datatypes_enabled"`
+	SampleFilter   []string        `toml:"samples_enabled"`
+	Log            telegraf.Logger `toml:"-"`
+
+	pmtBasePath            string
+	reader                 sourceReader
+	pmtTelemetryFiles      map[string]pmtFileInfo
+	pmtMetadata            *pmt
+	pmtAggregator          map[string]aggregator
+	pmtAggregatorInterface map[string]aggregatorInterface
+	pmtTransformations     map[string]map[string]transformation
+}
+
+// SampleConfig returns a sample configuration (See sample.conf).
+func (p *IntelPMT) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init performs one time setup of the plugin
+func (p *IntelPMT) Init() error {
+	err := p.checkPmtSpec()
+	if err != nil {
+		return err
+	}
+
+	err = p.explorePmtInSysfs()
+	if err != nil {
+		return fmt.Errorf("error while exploring pmt sysfs: %w", err)
+	}
+
+	return p.parseXMLs()
+}
+
+// Gather collects the plugin's metrics.
+func (p *IntelPMT) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+	var hasError atomic.Bool
+	for guid := range p.pmtTelemetryFiles {
+		wg.Add(1)
+		go func(guid string, fileInfo []fileInfo) {
+			defer wg.Done()
+			for _, info := range fileInfo {
+				data, err := os.ReadFile(info.path)
+				if err != nil {
+					hasError.Store(true)
+					p.Log.Errorf("Error occurred while gathering metrics: %v", err)
```

Comment:
Please use the accumulator to log the error. This way we can get additional statistics on errors... Same below
```suggestion
					acc.AddError(fmt.Errorf("gathering metrics failed: %v", err))
```